### ISSUE TITLE
Best way to handle expiring trial updates at stripe & webhooks

### ISSUE THREAD

**Issue by Tongboy** [payolapayments/payola#341]
Scenario:
in the stripe dashboard you hit 'expire trial now' to a users existing subscription. After the webhook runs the trial date is not updated in the payola record. If you instead update the value to any other 'real' value - the trial period is correctly updated.

Why:
the current `Payola::Subscription#sync_with!` only sets the `trial_end` (and other similar dates - ended_at, trial_start, & canceled_at) if the respective subscription value is present. Because the trial was expired Stripe sends a nil value in the webhook and payola ignore updating that value.

Proposed Fix:
change to `#sync_with!`
```ruby
  def sync_with!(stripe_sub)
    self.current_period_start = Time.at(stripe_sub.current_period_start)
    self.current_period_end   = Time.at(stripe_sub.current_period_end)
    self.ended_at             = stripe_sub.ended_at ? Time.at(stripe_sub.ended_at) : nil
    self.trial_start          = stripe_sub.trial_start ? Time.at(stripe_sub.trial_start) : nil
    self.trial_end            = stripe_sub.trial_end ? Time.at(stripe_sub.trial_end) : nil
    self.canceled_at          = stripe_sub.canceled_at ? Time.at(stripe_sub.canceled_at) : nil
    self.quantity             = stripe_sub.quantity
    self.stripe_status        = stripe_sub.status
    self.amount               = stripe_sub.plan.amount
    self.currency             = stripe_sub.plan.respond_to?(:currency) ? stripe_sub.plan.currency : Payola.default_currency
    self.cancel_at_period_end = stripe_sub.cancel_at_period_end
...
  end
```
basically - set the ended_at, trial_start, trial_end, & canceled_at values to nil if they value isn't present - otherwise do the appropriate time formatting for the present value.

Thoughts / Concerns?
